editable-list/main.js

```diff
@@ -1,132 +1,132 @@
+      super();
+
+      // attaches shadow tree and returns shadow root reference
+      // https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow
+      const shadow = this.attachShadow({ mode: 'open' });
+
+      // creating a container for the editable-list component
+      const editableListContainer = document.createElement('div');
+
+      // get attribute values from getters
+      const title = this.title;
+      const addItemText = this.addItemText;
+      const listItems = this.items;
+
+      // adding a class to our container for the sake of clarity
+      editableListContainer.classList.add('editable-list');
+
+      // creating the inner HTML of the editable list element
+      editableListContainer.innerHTML = `
+        <style>
+          li, div > div {
+            display: flex;
+            align-items: center;
+            justify-content: space-between;
+          }
+
+          .icon {
+            background-color: #fff;
+            border: none;
+            cursor: pointer;
+            float: right;
+            font-size: 1.8rem;
+          }
+        </style>
+        <h3>${title}</h3>
+        <ul class="item-list">
+          ${listItems.map(item => `
+            <li>${item}
+              <button class="editable-list-remove-item icon">&ominus;</button>
+            </li>
+          `).join('')}
+        </ul>
+        <div>
+          <label>${addItemText}</label>
+          <input class="add-new-list-item-input" type="text"></input>
+          <button class="editable-list-add-item icon">&oplus;</button>
+        </div>
+      `;
+
+      // binding methods
+      this.addListItem = this.addListItem.bind(this);
+      this.handleRemoveItemListeners = this.handleRemoveItemListeners.bind(this);
+      this.removeListItem = this.removeListItem.bind(this);
+
+      // appending the container to the shadow DOM
+      shadow.appendChild(editableListContainer);
+    }
 
-            if (textInput.value) {
-                const li = document.createElement('li');
-                const button = document.createElement('button');
-                const childrenLength = this.itemList.children.length;
+    // add items to the list
+    addListItem(e) {
+      const textInput = this.shadowRoot.querySelector('.add-new-list-item-input');
 
-                li.textContent = textInput.value;
-                button.classList.add('editable-list-remove-item', 'icon');
-                button.textContent = '\u2796';
+      if (textInput.value) {
+        const li = document.createElement('li');
+        const button = document.createElement('button');
+        const childrenLength = this.itemList.children.length;
 
-                this.itemList.appendChild(li);
-                this.itemList.children[childrenLength].appendChild(button);
+        li.textContent = textInput.value;
+        button.classList.add('editable-list-remove-item', 'icon');
+        button.innerHTML = '&ominus;';
```

Comment:
I noticed a bit of an inconsistency here. New list items were added with a different remove icon. Ideally, I like to double back and extract the rendering of the list to a function, so that all items are treated exactly the same.